packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -30,11 +31,20 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
   /// {@macro amplify_analytics_pinpoint_dart.amplify_analytics_pinpoint_dart}
   AmplifyAnalyticsPinpointDart({
     CachedEventsPathProvider? pathProvider,
+    LegacyNativeDataProvider? legacyNativeDataProvider,
+    DeviceContextInfoProvider? deviceContextInfoProvider,
     AppLifecycleProvider? appLifecycleProvider,
-    AnalyticsClient? analyticsClient,
-  })  : _appLifecycleProvider = appLifecycleProvider,
-        _pathProvider = pathProvider,
-        _analyticsClient = analyticsClient ?? AnalyticsClient();
+    SecureStorageFactory? secureStorageFactory,
+  })  : _pathProvider = pathProvider,
+        _legacyNativeDataProvider = legacyNativeDataProvider,
+        _deviceContextInfoProvider = deviceContextInfoProvider,
+        _appLifecycleProvider = appLifecycleProvider,
+        _secureStorageFactory = secureStorageFactory ??
+            ((scope) => AmplifySecureStorageWorker(
```

Comment:
The secure storage Dart package should have an equivalent factory method and that should be used here